pkg/sql/schemachanger/scbuild/internal/scbuildstmt/dependencies.go

```diff
@@ -329,3 +331,19 @@ type NameResolver interface {
 	// ResolveConstraint retrieves a constraint by name and returns its elements.
 	ResolveConstraint(relationID catid.DescID, constraintName tree.Name, p ResolveParams) ElementResultSet
 }
+
+type DescriptorIDGenerator interface {
+	GenerateUniqueDescID() catid.DescID
```

Comment:
This should be a method on the `BuilderState` interface. Its implementation could do whatever `MarkNewDescriptor` does so that that method never needs to be exposed in `scbuildstmt`. We really want to keep `scbuildstmt` as tight as possible and only have business logic that turns DDL statement ASTs into element targets, otherwise it very quickly becomes a huge mess.

Reply:
Totally agreed. Will do.

Reply:
change as suggested.